lib/mpl_toolkits/basemap/__init__.py

```diff
@@ -3901,6 +3901,9 @@ def drawlsmask(self,land_color="0.8",ocean_color="w",lsmask=None,
         # look for axes instance (as keyword, an instance variable
         # or from plt.gca().
         ax = kwargs.pop('ax', None) or self._check_ax()
+        # Clear saved lsmask is new lsmask is passed
+        if lsmask is not None or lsmask_lons is not None or lsmask_lats is not None:
+            self.lsmask = None
```

Comment:
I might do one final little check. I could see adding a check to see if the passed in `lsmask` is the same as the cached `lsmask`.